channeldb/control_tower.go

```diff
@@ -369,19 +357,22 @@ func nextPaymentSequence(tx *bbolt.Tx) ([]byte, error) {
 	return b, nil
 }
 
-// fetchPaymentStatus fetches the payment status from the bucket.  If the
-// status isn't found, it will default to "StatusUnknown".
+// fetchPaymentStatus fetches the payment status of the payment. If the payment
+// isn't found, it will default to "StatusUnknown".
 func fetchPaymentStatus(bucket *bbolt.Bucket) PaymentStatus {
-	// The default status for all payments that aren't recorded in
-	// database.
-	var paymentStatus = StatusUnknown
+	if bucket.Get(paymentSettleInfoKey) != nil {
+		return StatusSucceeded
+	}
+
+	if bucket.Get(paymentFailInfoKey) != nil {
+		return StatusFailed
+	}
 
-	paymentStatusBytes := bucket.Get(paymentStatusKey)
-	if paymentStatusBytes != nil {
-		paymentStatus.FromBytes(paymentStatusBytes)
+	if bucket.Get(paymentCreationInfoKey) != nil {
```

Comment:
Isn't this key always currently present?

Reply:
The first time calling `InitPayment`, it won't.